test/openssl/test_pkey.rb

```diff
@@ -38,6 +38,12 @@ def test_s_generate_parameters
     assert_raise(OpenSSL::PKey::PKeyError) {
       OpenSSL::PKey.generate_parameters("EC", "invalid" => "option")
     }
+  end
+
+  def test_s_generate_parameters_dsa
```

Comment:
```suggestion
  def test_s_generate_parameters_with_block
```

The purpose of this assertion is to check that the block is called if supplied, and an exception in it is handled correctly. It doesn't necessarily have to be DSA.

However, the other algorithm doesn't work in the FIPS mode either. I agree we have to skip this test case.

Reply:
All right. Fixed as your suggestion.